src/libraries/System.IO.Compression/tests/ZipArchive/zip_CreateTests.cs

```diff
@@ -30,115 +39,150 @@ public static void CreateModeInvalidOperations()
+        public static IEnumerable<object[]> Get_CreateNormal_Unseekable_Data()
+        {
+            foreach (string folder in _folderNames)
+            {
+                yield return new object[] { folder, false };
+                yield return new object[] { folder, true };
             }
         }
 
         [Theory]
-        [InlineData("small")]
-        [InlineData("normal")]
-        [InlineData("empty")]
-        [InlineData("emptydir")]
-        public static async Task CreateNormal_Unseekable(string folder)
+        [MemberData(nameof(Get_CreateNormal_Unseekable_Data))]
+        public static async Task CreateNormal_Unseekable(string folder, bool async)
         {
             using (var s = new MemoryStream())
             {
                 var testStream = new WrappedStream(s, false, true, false, null);
-                await CreateFromDir(zfolder(folder), testStream, ZipArchiveMode.Create);
-
-                IsZipSameAsDir(s, zfolder(folder), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true);
+                await CreateFromDir(zfolder(folder), testStream, async, ZipArchiveMode.Create);
+                await IsZipSameAsDir(s, zfolder(folder), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true, async);
             }
         }
 
-        [Fact]
-        public static async Task CreateNormal_Unicode_Seekable()
+        [Theory]
+        [MemberData(nameof(Get_Booleans_Data))]
+        public static async Task CreateNormal_Unicode_Seekable(bool async)
         {
             using (var s = new MemoryStream())
             {
                 var testStream = new WrappedStream(s, false, true, true, null);
-                await CreateFromDir(zfolder("unicode"), testStream, ZipArchiveMode.Create);
-
-                IsZipSameAsDir(s, zfolder("unicode"), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true);
+                await CreateFromDir(zfolder("unicode"), testStream, async, ZipArchiveMode.Create);
+                await IsZipSameAsDir(s, zfolder("unicode"), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true, async);
             }
         }
 
-        [Fact]
-        public static async Task CreateNormal_Unicode_Unseekable()
+        [Theory]
+        [MemberData(nameof(Get_Booleans_Data))]
+        public static async Task CreateNormal_Unicode_Unseekable(bool async)
         {
             using (var s = new MemoryStream())
             {
                 var testStream = new WrappedStream(s, false, true, false, null);
-                await CreateFromDir(zfolder("unicode"), testStream, ZipArchiveMode.Create);
-
-                IsZipSameAsDir(s, zfolder("unicode"), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true);
+                await CreateFromDir(zfolder("unicode"), testStream, async, ZipArchiveMode.Create);
+                await IsZipSameAsDir(s, zfolder("unicode"), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true, async);
```

Comment:
I was about to ask if we have a test that ensures that  sync and async methods produce the same outputs, but I assume that `IsZipSameAsDir` checks that?

Reply:
No, currently they don't, but I should write a test that checks the output of both is the same.

Reply:
I'd defer this to a future PR.

Reply:
Added two tests to verify this: https://github.com/dotnet/runtime/pull/114421/commits/cd1656e4ec36d5312bca1522796b805ceb1ad06f